test_scripts/Defects/4_5/1892_HB_from_SDL_after_receiving_HB_from_mobile.lua

```diff
@@ -0,0 +1,147 @@
+local mobile = require('mobile_connection')
+local tcp = require('tcp_connection')
+local file_connection = require('file_connection')
+
+--[[ Local Functions ]]
+local function BackUpIniFileAndSetHBValue()
+  commonPreconditions:BackupFile("smartDeviceLink.ini")
+  commonFunctions:write_parameter_to_smart_device_link_ini("HeartBeatTimeout", 5000)
+end
+
+local function RestoreIniFile()
+  commonPreconditions:RestoreFile("smartDeviceLink.ini")
+end
+
+local function OpenConnectionCreateSession(self)
+  config.defaultProtocolVersion = 3
+  -- self.heartbeatEnabled = false
+  local tcpConnection = tcp.Connection(config.mobileHost, config.mobilePort)
+  local fileConnection = file_connection.FileConnection("mobile.out", tcpConnection)
+  self.mobileConnection = mobile.MobileConnection(fileConnection)
+  self.mobileSession1= mobile_session.MobileSession(
+    self,
+    self.mobileConnection)
+  event_dispatcher:AddConnection(self.mobileConnection)
+  self.mobileSession1:ExpectEvent(events.connectedEvent, "Connection 1 started")
+  self.mobileConnection:Connect()
+  self.mobileSession1.sendHeartbeatToSDL = false
+  self.mobileSession1.answerHeartbeatFromSDL = false
+  self.mobileSession1.ignoreHeartBeatAck = false
+  self.mobileSession1:StartService(7)
+end
+
+local function RegisterAppInterface(self)
+  local CorIdRegister = self.mobileSession1:SendRPC("RegisterAppInterface", {
+      syncMsgVersion ={
+        majorVersion = 4,
+        minorVersion = 3
+      },
+      appName = config.application1.registerAppInterfaceParams.appName,
+      isMediaApplication = true,
+      languageDesired = 'EN-US',
+      hmiDisplayLanguageDesired = 'EN-US',
+      appHMIType = { "DEFAULT" },
+      appID = config.application1.registerAppInterfaceParams.appID
+    })
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered",
+    { application = {appName = config.application1.registerAppInterfaceParams.appName }})
+  self.mobileSession1:ExpectResponse(CorIdRegister, { success = true, resultCode = "SUCCESS" })
+  self.mobileSession1:ExpectNotification("OnHMIStatus",
+    { systemContext = "MAIN", hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE" })
+end
+
+local function ExpectationAfterAppRegistration(self)
+  local hmiAppId = commonDefects.getHMIAppId(1)
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered",
+    {unexpectedDisconnect = true, appID = hmiAppId})
+  :Times(0)
+  commonFunctions:userPrint (33,"Log: AppSession started, HB disabled")
+  commonFunctions:userPrint (33, "Log: App v.3 disconnection not expected since no HB ACK and timer" ..
+  "should be started by SDL till the HB request from app first")
+  local HBEvent = events.Event()
+  HBEvent.matches =
+  function(_, data)
+    return data.frameType == 0 and
+    data.serviceType == 0 and
+    (data.sessionId == self.mobileSession1.sessionId) and
+    data.frameInfo == 0
+  end
+  self.mobileSession1:ExpectEvent(HBEvent, "HB")
+  :Times(0)
+  commonDefects.delayedExp(10000)
```

Comment:
@GetmanetsIrina here should be lesser wait value because ATF has its own heartbeat monitor, which will shut down session on timeout and triggers SDL to send OnAppUnregistered, which breaks this test. 
Propose to replace this timeout with: `delayedExp(config.heartbeatTimeout - 1000)` as it is an ATF heartbeat monitor time, during which SDL can send nothing.

Reply:
@AKalinich-Luxoft , according self.mobileSession1.sendHeartbeatToSDL = false also added ATF should not start its own HB. In case of updating expectation to config.heartbeatTimeout - 1000 or config.heartbeatTimeout sometimes tests case does not catch unexpected HB from SDL. For now ATF has issue regarding HB.